### PR TITLE
Gmdx 117

### PR DESCRIPTION
**Pull request by bjdupuis**
Getting the Jenkinsfile working to:

- bump version in package.json
- publish to public NPM repo
- push a git tag with the version
- mail the email list on failure and recovery

### CODE REVIEW

**Review by crumpf — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+          sh '''
+              echo "registry=https://registry.npmjs.org" > ./.npmrc
+              echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" >> ./.npmrc
+          '''
+          sh "${env.WORKSPACE}/${env.NPM_UTIL_PATH}/scripts/auto-version-bump.sh"
+          
+          // Do not include the npm-utils directory or the publish credentials in the published package.
+          sh '''
+              echo "npm-utils" >> .npmignore
+              npm publish 1>&2
```
Comment (ElizaSapir) [inline Jenkinsfile]: We should first run ` npm publish --dry-run` to see there are no errors and then ` npm publish`

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  options {
+    quietPeriod(480)
+    disableConcurrentBuilds()
+  }
+
+  environment {
+    NPM_UTIL_PATH = "npm-utils"
+    REPO_DIR = "repo"
+    SHORT_BRANCH = env.GIT_BRANCH.replaceFirst(/^origin\//, '');
+    NPM_TOKEN = credentials('2844c47b-19b8-4c5f-b901-190de49c0883')
```
Comment (ElizaSapir) [inline Jenkinsfile]: Can it be exposed? It should be a parameter we set on Jenkins environment while running this.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+          ])
+        }
+      }
+    }
+    stage('Import notifications lib') {
+      steps {
+        script {
+          // clone pipelines repo
+          dir('pipelines') {
+            git branch: 'master',
```
Comment (AfanasievAnton) [inline Jenkinsfile:53]: I might be wrong, but I think we do not have git branch `master` in this repo.

File: Jenkinsfile (hunk 2 above)
Comment (bjdupuis) [reply Jenkinsfile]: I can expose the credentials id as a parameter, yes. I can't expose the credentials themselves.

File: Jenkinsfile (hunk 1 above)
Comment (bjdupuis) [reply Jenkinsfile]: Given requested changes I'll add a dry-run back.

File: Jenkinsfile (hunk 3 above)
Comment (bjdupuis) [reply Jenkinsfile:53]: @AfanasievAnton this isn't pulling code from our repo but from the `pipeline-library` repo. Similar below with `npm-utils`.

**Review by AfanasievAnton — APPROVED** [review]

**Comment by bjdupuis** [comment]
Trying to get deployment credentials parameterized. Will reopen when complete.

**Pull request closed without merging**